src/snowflake/snowpark/mock/_functions.py

```diff
@@ -929,3 +931,61 @@ def mock_to_variant(expr: ColumnEmulator):
     res = expr.copy()
     res.sf_type = ColumnType(VariantType(), expr.sf_type.nullable)
     return res
+
+
+def cast_to_datetime(date):
+    return datetime.datetime.fromordinal(date.toordinal())
+
+
+def add_years(date, duration):
+    return date.replace(year=date.year + duration)
+
+
+def add_months(scalar, date, duration):
+    # Keep months in 1..12
+    month = (date.month - 1 + scalar * duration) % 12 + 1
+
+    # Add years as needed
+    year = date.year + (date.month - 1 + scalar * duration) // 12
+
+    return date.replace(year=year, month=month)
+
+
+def add_timedelta(unit, date, duration, scalar=1):
+    return date + datetime.timedelta(**{f"{unit}s": duration * scalar})
+
+
+@patch("dateadd")
+def mock_dateadd(part: str, expr1: ColumnEmulator, expr2: ColumnEmulator):
+    # Extract a standardized name
+    part = unalias_datetime_part(part)
+    sf_type = expr2.sf_type
+    ts_type = ColumnType(TimestampType(TimestampTimeZone.NTZ), expr2.sf_type.nullable)
+
+    def nop(x):
```

Comment:
NIT: `cast = lambda x : x  # noop`?

Reply:
precommit fails when assigning lambdas to variables.